### PR TITLE
ci: add yamllint rules

### PR DESCRIPTION
**Pull request by sakulali**
### Type of change:
fix: #1062 
Add `yamllint` rules check `samples/deploy/` directory

- [ ] Bugfix
- [ ] New feature provided
- [ ] Improve performance
- [ ] Backport patches
- [ ] Documentation
- [ ] Refactor
- [ ] Chore
- [x] CI/CD or Tests

### Pre-submission checklist:

- [ ] Did you explain what problem does this PR solve? Or what new features have been added?
- [ ] Have you added corresponding test cases?
- [ ] Have you modified the corresponding document?
- [ ] Is this PR backward compatible? **If it is not backward compatible, please discuss on the [mailing list](https://github.com/apache/apisix-ingress-controller#community) first**

### CODE REVIEW

File: .github/workflows/yamllint-checker.yml [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+      - run: pip install yamllint==1.29.0
+      - name: 🚀 Run yamllint
+        run: yamllint samples/deploy  # focus on `samples/deploy` yaml files directory
+
+      # Run GitHub Actions locally (nektos/act)
+      # - name: 🚀 Run yamllint
+      #   run: |
+      #     make verify-yamllint
+
+
```
Comment (tao12345666333) [inline .github/workflows/yamllint-checker.yml:68]: remove these lines

File: .yamllint.yml [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+  commas: disable  # default: enable
+  comments: disable  # default: {level: warning}
+  comments-indentation: disable  # default: {level: warning}
+  document-end: disable
+  document-start: disable  # default: {level: warning}
+  empty-lines: enable
+  empty-values: disable
+  float-values: disable
+  hyphens: enable
+  indentation: disable  # default: enable
```
Comment (tao12345666333) [inline .yamllint.yml:39]: Why disabled it?

Comment (tao12345666333) [reply .yamllint.yml:39]: @sakulali we should enable these options and fix the format of YAML. Thanks

Comment (tao12345666333) [reply .yamllint.yml]: don't worry. It doesn't have any real impact on users.

File: Makefile [hunk 3]
```diff
@@ -222,9 +222,19 @@ verify-license:
...
+
 ### verify-all:           Verify all verify- rules.
 .PHONY: verify-all
-verify-all: verify-codegen verify-license verify-mdlint
+verify-all: verify-codegen verify-license verify-mdlint verify-yamllint
+
+### update-yamlfmt:       Update yaml files format for `samples/deploy` directory.
+.PHONY: update-yamlfmt
+update-yamlfmt:
+	go install github.com/google/yamlfmt/cmd/yamlfmt@latest && yamlfmt samples/deploy
```
Comment (tao12345666333) [inline Makefile:237]: The `peterdavehello/yamllint ` image is Dockerized https://github.com/adrienverge/yamllint

But in the `update-yamlfmt` job you install https://github.com/google/yamlfmt, why?

**Review by tao12345666333 — COMMENTED** [review]
just one question

**Review by tao12345666333 — APPROVED** [review]
LGTM

File: Makefile (hunk 3 above)
Comment (tao12345666333) [reply Makefile:237]: Yes, I know, so I approved this PR

**Review by lingsamuel — APPROVED** [review]

**Pull request merged**